pydantic/_internal/_std_types_schema.py

```diff
@@ -217,22 +221,46 @@ def path_schema_prepare_pydantic_annotations(
     }:
         return None
 
+    is_path_like_subtype_invalid = (
+        orig_source_type == os.PathLike
+        and orig_source_type_arg
+        and orig_source_type_arg
+        not in {
+            str,
+            bytes,
+            Any,
+        }
+    )
+
+    if is_path_like_subtype_invalid:
+        return None
+
     metadata, remaining_annotations = _known_annotated_metadata.collect_known_metadata(annotations)
-    _known_annotated_metadata.check_metadata(metadata, _known_annotated_metadata.STR_CONSTRAINTS, source_type)
+    _known_annotated_metadata.check_metadata(metadata, _known_annotated_metadata.STR_CONSTRAINTS, orig_source_type)
+
+    construct_path = orig_source_type
```

Comment:
What about:

```py
construct_path = pathlib.PurePath if orig_source_type is os.PathLike else orig_source_type
constrained_schema = (
    core_schema.bytes_schema(**metadata) if source_type_args and source_type_args[0] is bytes 
    else core_schema.str_schema(**metadata)
)

def path_validator(input_value: str) -> os.PathLike[Any]:
    try:
        return construct_path(input_value)
    except TypeError as e:
        raise PydanticCustomError('path_type', 'Input is not a valid path') from e
```

Reply:
I realize I've skipped the decoding here, one moment... trying to add back in with correct typing

Reply:
I think this is type safe:

```py
def path_validator(input_value: str | bytes) -> os.PathLike[str]:
    try:
        if source_type_args and source_type_args[0] is bytes:
            if isinstance(input_value, bytes):
                try:
                    input_value = input_value.decode()
                except UnicodeDecodeError as e:
                    raise PydanticCustomError('bytes_type', 'Input must be valid bytes') from e
            else:
                raise PydanticCustomError('bytes_type', 'Input must be bytes')
        elif not isinstance(input_value, str):
            raise PydanticCustomError('path_type', 'Input is not a valid path')
    
        return construct_path(input_value)
    except TypeError as e:
        raise PydanticCustomError('path_type', 'Input is not a valid path') from e
```